### PR TITLE
Remove `TextBox::SetTimerInfo`, pass to the constructor instead

### PR DESCRIPTION
**Pull request by dpallotti**
#### Remove `TextBox::SetTimerInfo`, pass to the constructor instead
`TimerChain` needs the added `TextBox` to already have the `TimerInfo` set, to
correctly update the `min_timestamp_` and `max_timestamp` of a `TimerBlock`.
Also addressing some lint warnings as there's too much yellow in my IDE.

Bug: http://b/188375720

Test: Capture Trata with Vulkan layer and instrumented functions: notice that
scheduler and GPU tracks are now filled. Also enable a frame track and notice
that it also now works, although sometimes affected by the unrelated crash
http://b/188521605.
#### Delete copy- and move- assignment of `TextBox`
As discussed, so that elements of `TimerChain` cannot be reassigned.

Bug: http://b/188375720

### CODE REVIEW

**Review by beckerhe — APPROVED** [review]
LGTM. Thanks a lot, Davide!

**Review by florian-kuebler — APPROVED** [review]
LGTM, Thanks!

File: src/OrbitGl/CaptureStatsTest.cpp [hunk 1]
```diff
@@ -10,18 +10,14 @@
 #include "TextBox.h"
 
 TextBox* CreateScope(int32_t pid, int32_t tid, int32_t cpu, uint64_t start, uint64_t end) {
-  const size_t kNumScopes = 1024;
-  static TextBox scope_buffer[kNumScopes];
-  static size_t index = 0;
-  EXPECT_TRUE(index < kNumScopes - 2);
```
Comment (karupayun) [inline src/OrbitGl/CaptureStatsTest.cpp:16]: I know that we are inside a test, but we are not having this invariant anymore or simply we are not checking it?

**Review by karupayun — APPROVED** [review]
LGTM! Thanks Davide!

File: src/OrbitGl/CaptureStatsTest.cpp (hunk 1 above)
Comment (dpallotti) [reply src/OrbitGl/CaptureStatsTest.cpp:16]: As far as I understand, this was just a limit posed by using an array, but a vector can just be used instead.

Comment (dpallotti) [reply src/OrbitGl/CaptureStatsTest.cpp:16]: At the same time now a test fails on Windows in this file, so let's see.

Comment (dpallotti) [reply src/OrbitGl/CaptureStatsTest.cpp:16]: Rearranged this code a bit and switched to a list.

**Comment by pierricgimmig** [comment]
Thanks @dpallotti !

**Pull request merged**